Fix color-contrast a11y failures in Discussions

bjacobel opened the pull request:
[TNL-4639](https://openedx.atlassian.net/browse/TNL-4639)
[TNL-4644](https://openedx.atlassian.net/browse/TNL-4644)
[TNL-4635](https://openedx.atlassian.net/browse/TNL-4635)

- [ ] Code review: @andy-armstrong 
- [x] Code review: @alisan617

andy-armstrong commented on lms/static/sass/discussion/_discussion.scss:
```diff
@@ -444,7 +444,7 @@ section.discussion-pagination {
     display: block;
     padding: ($baseline/2) 0;
-    color: $gray;
+    color: $gray-d2;
```
Can you change this to use a variable from variables-v1.scss (and variables-v2.scss), and if there isn't a close enough one then introduce a new variable. My goal is to have no colors used directly in the Sass, but to instead have everything defined through variables that can be overridden in a theme.

andy-armstrong commented on lms/static/sass/discussion/_mixins.scss:
```diff
@@ -61,7 +61,7 @@
   padding-top: 3px;
   width: 100%;
-  color: $gray-l2;
+  color: $gray-d2;
```
This should use the same variable as with my other comment.

andy-armstrong commented on lms/static/sass/course/layout/_courseware_header.scss:
```diff
@@ -42,19 +42,14 @@
           &:hover,
           &:focus {
-            color: $uxpl-blue-base;
+            color: $uxpl-blue-hover-active;
```
I'm a bit confused here. Isn't this the same color that is defined in the parent rule? Does this mean that there isn't a color change on hover/focus?

bjacobel replied:
Shit, yeah. I tried to collapse this Sass down to avoid some repetition but now I'm just applying the same color all the time.

andy-armstrong replied:
Are you going to introduce a different color? If not, can you remove the redundant rules?

bjacobel commented:
Updated to use themeable variables for colors.

bjacobel commented:
jenkins run a11y

bjacobel commented:
Updated again to remove some duplicated sass. Spun up a sandbox - things worth checking out include the course tab navigation - I'm not sure if I fixed a bug, or removed a feature (the `active` and `hover` colors in prod are slightly different on non-pattern-library pages, and I've made them the same). 

(old behavior)
![](https://i.bjacobel.com/20160914-t6hj3.png)

alisan617 commented:
👍

bjacobel commented:
jenkins run a11y

andy-armstrong commented:
👍 LGTM. I don't understand why the a11y test is failing :-(

alisan617 commented:
This list looks broken in IE10. That might be the case for the old FF in jenkins. There is text with white color. That's how a11y fails.
<img width="728" alt="screen shot 2016-09-16 at 11 15 38 am" src="https://cloud.githubusercontent.com/assets/16127998/18591058/1341f4b2-7bff-11e6-836e-17b734d79139.png">

bjacobel commented:
We can't figure out why these are failing on Jenkins and not locally, so I've set the affected tests back to being ignored for now in 48bec13d57ae830eda24bd14f637645fcd5dd0ce.

bjacobel commented:
@alisan617 Everything looks fine if I manually load up the Firefox version that Devstack and Jenkins use, though. We don't support IE10 so I'm not super worried about it being broken there.

![](https://i.bjacobel.com/20160916-bcdw7.png)

bjacobel commented:
jenkins run js
